src/main/java/bio/terra/service/filedata/azure/tables/AzureTableUtils.java

```diff
@@ -0,0 +1,66 @@
+package bio.terra.service.filedata.azure.tables;
+
+import bio.terra.service.filedata.exception.FileSystemExecutionException;
+import bio.terra.service.filedata.google.firestore.ApiFutureGenerator;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AzureTableUtils {
+  private final Logger logger = LoggerFactory.getLogger(AzureTableUtils.class);
+  private static final int SLEEP_BASE_SECONDS = 1;
+  private static final int AZURE_STORAGE_RETRIES = 1;
+
+  @Autowired
+  public AzureTableUtils() {}
+
+  <V> void scanTableObjects(TableClient tableClient, ApiFutureGenerator<V, TableEntity> generator) {
+    ListEntitiesOptions options = new ListEntitiesOptions();
+    PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
+    entities.stream()
+        .forEach(
+            entity -> {
+              try {
+                retryOperation(entity, generator);
```

Comment:
The SDK provides retriable capabilities out of the box. The client builder allows you to set an appropriate policy and retry options. More details [here](https://docs.microsoft.com/en-us/java/api/com.azure.data.tables.tableserviceclientbuilder.retrypolicy?view=azure-java-stable#com_azure_data_tables_TableServiceClientBuilder_retryPolicy_com_azure_core_http_policy_RetryPolicy_). 

However, having another retry policy is okay if there is a need to retry due to an expected functional behavior no covered by the SDK. In such a case, a retry policy must be triggered by well-defined exceptions that identify the condition. Ultimately, the goal is to avoid duplicate retries attempts.

Reply:
Thanks! That's good to know